tests/Fixer/Whitespace/NoSpacesAroundOffsetFixerTest.php

```diff
@@ -0,0 +1,325 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Whitespace;
+
+use PhpCsFixer\Test\AbstractFixerTestCase;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ *
+ * @internal
+ */
+final class NoSpacesAroundOffsetFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideInsideCases
+     */
+    public function testFixSpaceInsideOffset($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @dataProvider provideOutsideCases
+     */
+    public function testFixSpaceOutsideOffset($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function testLeaveNewLinesAlone()
+    {
+        $expected = <<<'EOF'
+<?php
+
+class Foo
+{
+    private function bar()
+    {
+        if ([1, 2, 3] && [
+            'foo',
+            'bar' ,
+            'baz'// a comment just to mix things up
+        ]) {
+            return 1;
+        };
+    }
+}
+EOF;
+        $this->doTest($expected);
+    }
+
+    public function testLeaveCommentsAlone()
+    {
+        $expected = <<<'EOF'
+<?php
+
+$withComments[0] // here is a comment
+    [1] // and here is another
+    [2] = 3;
+EOF;
+        $this->doTest($expected);
+    }
+
+    public function testLeaveComplexString()
+    {
+        $expected = <<<'EOF'
+<?php
+
+echo "I am printing some spaces here    {$foo->bar[1]}     {$foo->bar[1]}.";
+EOF;
+        $this->doTest($expected);
+    }
+
+    public function testLeaveFunctions()
+    {
+        $expected = <<<'EOF'
+<?php
+
+function someFunc()    {   $someVar = [];   }
+EOF;
+        $this->doTest($expected);
+    }
+
+    public function provideOutsideCases()
+    {
+        return array(
+            array(
```

Comment:
I check, so it is more like `inside` and `before` (rather than `outside`), all good :+1: